common/src/main/java/org/opensearch/ml/common/connector/functions/preprocess/CohereMultiModalEmbeddingPreProcessFunction.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.common.connector.functions.preprocess;
+
+import static org.opensearch.ml.common.utils.StringUtils.convertScriptStringToJsonString;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.opensearch.ml.common.dataset.TextDocsInputDataSet;
+import org.opensearch.ml.common.dataset.remote.RemoteInferenceInputDataSet;
+import org.opensearch.ml.common.input.MLInput;
+
+public class CohereMultiModalEmbeddingPreProcessFunction extends ConnectorPreProcessFunction {
+
+    public CohereMultiModalEmbeddingPreProcessFunction() {
+        this.returnDirectlyForRemoteInferenceInput = true;
+    }
+
+    @Override
+    public void validate(MLInput mlInput) {
+        validateTextDocsInput(mlInput);
+        List<String> docs = ((TextDocsInputDataSet) mlInput.getInputDataset()).getDocs();
+        if (docs.isEmpty() || (docs.size() == 1 && docs.getFirst() == null)) {
+            throw new IllegalArgumentException("No image provided");
+        }
+    }
+
+    @Override
+    public RemoteInferenceInputDataSet process(MLInput mlInput) {
+        TextDocsInputDataSet inputData = (TextDocsInputDataSet) mlInput.getInputDataset();
+        Map<String, String> parametersMap = new HashMap<>();
+        parametersMap.put("images", inputData.getDocs().getFirst());
```

Comment:
Only support image input? Should we also consider support text input ?

Reply:
Cohere multi-modal doesn't support text with image input. Either image input or text input. [Example notebook](https://github.com/cohere-ai/notebooks/blob/main/notebooks/Multimodal_Semantic_Search.ipynb)

For text input, we could use our regular one: `connector.pre_process.cohere.embedding`

Reply:
Just one per request.

Reply:
An array of image data URIs for the model to embed. Maximum number of images per call is 1.

Reply:
Got it, suggest add more java doc to explain these details